conan/tools/gnu/gnutoolchain.py

```diff
@@ -0,0 +1,280 @@
+
+        self.cppstd = cppstd_flag(self._conanfile)
+        self.arch_flag = architecture_flag(self._conanfile.settings)
+        self.libcxx, self.gcc_cxx11_abi = libcxx_flags(self._conanfile)
+        self.fpic = self._conanfile.options.get_safe("fPIC")
+        self.msvc_runtime_flag = self._get_msvc_runtime_flag()
+        self.msvc_extra_flags = self._msvc_extra_flags()
+
+        # Cross build triplets
+        self.cross_build = {
+            "host": {"triplet": self._conanfile.conf.get("tools.gnu:host_triplet")},
+            "build": {"triplet": self._conanfile.conf.get("tools.gnu:build_triplet")}
+        }
+        is_cross_building = cross_building(self._conanfile)
+        if is_cross_building:
+            compiler = self._conanfile.settings.get_safe("compiler")
+            # Host triplet
+            if not self.cross_build["host"]["triplet"]:
+                os_host = conanfile.settings.get_safe("os")
+                arch_host = conanfile.settings.get_safe("arch")
+                self.cross_build["host"] = _get_gnu_triplet(os_host, arch_host, compiler=compiler)
+            # Build triplet
+            if not self.cross_build["build"]["triplet"]:
+                os_build = conanfile.settings_build.get_safe('os')
+                arch_build = conanfile.settings_build.get_safe('arch')
+                self.cross_build["build"] = _get_gnu_triplet(os_build, arch_build, compiler=compiler)
+
+        sysroot = self._conanfile.conf.get("tools.build:sysroot")
+        sysroot = sysroot.replace("\\", "/") if sysroot is not None else None
+        self.sysroot_flag = "--sysroot {}".format(sysroot) if sysroot else None
+        self.configure_args = {}
+        self.autoreconf_args = {"--force": None, "--install": None}
+        self.make_args = {}
+        # Initializing configure arguments: triplets, shared flags, dirs flags, etc.
+        self.configure_args.update(self._get_default_configure_shared_flags())
+        self.configure_args.update(self._get_default_configure_install_flags())
+        self.configure_args.update(self._get_default_triplets())
+        # Apple stuff
+        is_cross_building_osx = (is_cross_building
+                                 and conanfile.settings_build.get_safe('os') == "Macos"
+                                 and is_apple_os(conanfile))
+        min_flag, arch_flag, isysroot_flag = (
+            resolve_apple_flags(conanfile, is_cross_building=is_cross_building_osx)
+        )
+        # https://man.archlinux.org/man/clang.1.en#Target_Selection_Options
+        self.apple_arch_flag = arch_flag
+        # -isysroot makes all includes for your library relative to the build directory
+        self.apple_isysroot_flag = isysroot_flag
+        self.apple_min_version_flag = min_flag
+        # MSVC common stuff
+        self._initialize_default_extra_env()
+
+    def yes_no(self, option_name, default=None, negated=False):
```

Comment:
Simple function wrapper, but it's very common to see it defined all over the recipes using AutotoolsToolchain

Reply:
Have you considered simply directly mapping `True`/`False` to `"yes"`/`"no"` in the generator itself as an alternative? It would look even cleaner and would be more consistent with CMakeToolchain, imo. The only slight downside is that `None` from `self.options.get_safe()` would map to no parameter, which usually gets interpreted as a positive value.